docs/API.md

```diff
@@ -1193,15 +1193,20 @@ build caches. More specifically:
 
 ### `craneLib.overrideToolchain`
 
-`overrideToolchain :: drv -> set`
+`overrideToolchain :: (set -> drv) -> set`
+`overrideToolchain :: drv -> set` (deprecated)
```

Comment:
Do we want to deprecate this? Imo it's perfectly fine to use the old form when not cross-compiling and we do have a warning in place if you try to do this with a cross nixpkgs.

Though I suppose if you don't care (or your toolchain is not in scope of `pkgs`) you can just as well do:
```
overrideToolchain (_: myToolchain)
```

Reply:
I guess it's not so much "deprecated" as "discouraged"? As in using the function-method definitely works in all cases, while using a direct value only works if you don't care about cross-compiling (though if you suddenly started to care, our warning would catch that).

I'll rename this to `legacy` so it doesn't carry as heavy as a connotation!